File: src/data_structure/trees/bst_insert.cpp
/*
Node is defined as 

typedef struct node
{
   int data;
   node * left;
   node * right;
}node;

*/


node * insert(node * root, int value)
{
    node * n = (struct node *) malloc (sizeof(struct node));
    n->data = value;
    n->left = NULL;
    n->right = NULL;
    if (root == NULL)
    {
        return n;
    }
    if (root->data > value) {
        if (root->left == NULL)
        {
            root->left = n;
            return NULL;
        } 
        else 
        {
            insert(root->left, value);
        }
    } else {
        if (root->right == NULL)
        {
            root->right = n;
            return NULL;
        } 
        else 
        {
            insert(root->right, value);
        }
    }


   return root;
}

File: src/data_structure/trees/bst_insert.java
 /* Node is defined as :
 class Node 
    int data;
    Node left;
    Node right;
    
    */

static Node Insert(Node root,int value) {
    if (root == null) {
        Node n = new Node();
        n.data = value;
        n.left = null;
        n.right = null;
        return n;
    }
    if (root.data > value) {
        if (root.left == null) {
            Node n = new Node();
            n.data = value;
            n.left = null;
            n.right = null;
            root.left = n;
        } else {
            Insert(root.left, value);
        }
    } else {
        if (root.right == null) {
            Node n = new Node();
            n.data = value;
            n.left = null;
            n.right = null;
            root.right = n;
        } else {
            Insert(root.right, value);
        }
    }
    return root;
}
